Firestore/Source/Local/FSTLocalStore.mm

```diff
@@ -390,9 +390,9 @@ - (void)notifyLocalViewChanges:(NSArray<FSTLocalViewChanges *> *)viewChanges {
 }
 
 - (nullable FSTMutationBatch *)nextMutationBatchAfterBatchID:(FSTBatchID)batchID {
-  FSTWriteGroup *group = [self.persistence startGroupWithAction:@"nextMutationBatchAfterBatchID"];
-  FSTMutationBatch *result = [self.mutationQueue nextMutationBatchAfterBatchID:batchID];
-  [self.persistence commitGroup:group];
+  FSTMutationBatch *result = self.persistence.run([&]() -> FSTMutationBatch * {
```

Comment:
It does appear possible. We capture `self`, but it looks like ARC will do the right thing capturing `self` by value. Since we are not saving the lambdas anywhere, we are not in danger of creating a retain cycle. 

All that being said, is there an advantage to by value instead of by reference, given that we are not saving the lambda?

Reply:
Entirely optional:
* for a lambda taking no arguments, empty `()` braces can be omitted;
* return type can also be omitted.

Feel free to keep either/both if you find that form more readable.